src/core/Akka.Remote/MessageSerializer.cs

```diff
@@ -41,16 +42,16 @@ public static object Deserialize(ActorSystem system, SerializedMessage messagePr
         /// <returns>SerializedMessage.</returns>
         public static SerializedMessage Serialize(ActorSystem system, Address address, object message)
         {
-            Serializer serializer = system.Serialization.FindSerializerFor(message);
+            var objectType = message.GetType();
+            Serializer serializer = system.Serialization.FindSerializerForType(objectType);
 
             var serializedMsg = new SerializedMessage
             {
                 Message = ByteString.CopyFrom(serializer.ToBinaryWithAddress(address, message)),
```

Comment:
Also `ByteString.CopyFrom` make a copy of an array. But unfortunately `Google.Protobuf` does not have `CopyFromUnsafe`. Actually it has this feature, but it is internal https://github.com/google/protobuf/blob/master/csharp/src/Google.Protobuf/ByteString.cs#L66